modules/standard/IO.chpl

```diff
@@ -3006,7 +3079,51 @@ private inline proc EFORMAT do return chpl_macro_int_EFORMAT():c_int;
 
 
 pragma "no doc"
-proc _channel._ch_ioerror(error:errorCode, msg:string) throws {
+proc fileReader._ch_ioerror(error:errorCode, msg:string) throws {
+  var path:string = "unknown";
```

Comment:
Yeah, that was at least partially in the interest of time

Reply:
I was also uncertain how much of it could be specialized for the reading or writing case.  By having a strict separation, it could be that there's room for optimization or adjustments that would make sense for one but not the other.